cli/azd/pkg/project/ai_helper.go

```diff
@@ -0,0 +1,713 @@
+	ctx context.Context,
+	scope *ai.Scope,
+	serviceConfig *ServiceConfig,
+	config *ai.ComponentConfig,
+) (*armmachinelearning.EnvironmentVersion, error) {
+	yamlFilePath := filepath.Join(serviceConfig.Path(), config.Path)
+	_, err := os.Stat(yamlFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	environmentName, err := config.Name.Envsubst(a.env.Getenv)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing environment name value: %w", err)
+	}
+
+	if environmentName == "" {
+		environmentName = fmt.Sprintf("%s-environment", serviceConfig.Name)
+	}
+
+	nextVersion := "1"
+	envContainerResponse, err := a.envContainersClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		scope.Workspace(),
+		environmentName,
+		nil,
+	)
+	if err == nil {
+		nextVersion = *envContainerResponse.Properties.NextVersion
+	}
+
+	environmentArgs := []string{
+		"-t", "environment",
+		"-s", scope.SubscriptionId(),
+		"-g", scope.ResourceGroup(),
+		"-w", scope.Workspace(),
+		"-f", yamlFilePath,
+		"--set", fmt.Sprintf("name=%s", environmentName),
+		"--set", fmt.Sprintf("version=%s", nextVersion),
+	}
+
+	environmentArgs, err = a.applyOverrides(environmentArgs, config.Overrides)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := a.pythonBridge.Run(ctx, ai.MLClient, environmentArgs...); err != nil {
+		return nil, err
+	}
+
+	envVersionResponse, err := a.envVersionsClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		scope.Workspace(),
+		environmentName,
+		nextVersion,
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	a.env.DotenvSet(AiEnvironmentEnvVarName, environmentName)
+
+	return &envVersionResponse.EnvironmentVersion, nil
+}
+
+// CreateModelVersion creates a new model version
+func (a *aiHelper) CreateModelVersion(
+	ctx context.Context,
+	scope *ai.Scope,
+	serviceConfig *ServiceConfig,
+	config *ai.ComponentConfig,
+) (*armmachinelearning.ModelVersion, error) {
+	yamlFilePath := filepath.Join(serviceConfig.Path(), config.Path)
+	_, err := os.Stat(yamlFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	modelName, err := config.Name.Envsubst(a.env.Getenv)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing model name value: %w", err)
+	}
+
+	if modelName == "" {
+		modelName = fmt.Sprintf("%s-model", serviceConfig.Name)
+	}
+
+	nextVersion := "1"
+	modelContainerResponse, err := a.modelContainersClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		scope.Workspace(),
+		modelName,
+		nil,
+	)
+	if err == nil {
+		nextVersion = *modelContainerResponse.Properties.NextVersion
+	}
+
+	modelArgs := []string{
+		"-t", "model",
+		"-s", scope.SubscriptionId(),
+		"-g", scope.ResourceGroup(),
+		"-w", scope.Workspace(),
+		"-f", yamlFilePath,
+		"--set", fmt.Sprintf("name=%s", modelName),
+		"--set", fmt.Sprintf("version=%s", nextVersion),
+	}
+
+	modelArgs, err = a.applyOverrides(modelArgs, config.Overrides)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := a.pythonBridge.Run(ctx, ai.MLClient, modelArgs...); err != nil {
```

Comment:
does the operation takes some time?
Wondering if we want to pass here a showProgress screen (like the one from docker build) to show what's happening

Reply:
There is not really any interesting output that would be valuable to the user.  Some of the operations just take time to complete.